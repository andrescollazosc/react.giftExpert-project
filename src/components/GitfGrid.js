import React from "react";
import { useFetchGifts } from "../hooks/useFetchGifts";
// import { getGift } from "../helpers/getGifts";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GitfGrid = ({ category }) => {
  // const [images, setImages] = useState([]);
  const { data, loading } = useFetchGifts(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading ?? <p className="animate__animated animate__fadeIn">Loading</p>}

      <div className="card-grid">
        {data.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GitfGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
