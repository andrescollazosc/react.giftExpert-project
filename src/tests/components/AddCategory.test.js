import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";
import React from "react";
import "@testing-library/jest-dom";

describe("Name of the group", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("Must be show succesfully ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Must be change textbox", () => {
    const input = wrapper.find("input");
    const value = "Hello world";
    input.simulate("change", { target: { value } });

    expect(wrapper.find("p").text().trim()).toBe(value);
  });

  test("Should not post the information submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test("Must be call the setCategories and clean the textbox", () => {
    const value = "Hello world";
    wrapper.find("input").simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
