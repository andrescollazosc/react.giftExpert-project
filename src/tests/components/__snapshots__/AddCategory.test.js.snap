// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group Must be show succesfully  1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <div
    className="container-search"
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
</form>
`;
